x/concentrated-liquidity/msg_server.go

```diff
@@ -19,6 +21,46 @@ func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
 	}
 }
 
+func NewMsgCreatorServerImpl(keeper *Keeper) concentrated.MsgCreatorServer {
+	return &msgServer{
+		keeper: keeper,
+	}
+}
+
+// CreatePool attempts to create a pool returning the newly created pool ID or an error upon failure.
+// The pool creation fee is used to fund the community pool.
+// It will create a dedicated module account for the pool and sends the initial liquidity to the created module account.
+func (server msgServer) CreatePool(goCtx context.Context, msg swaproutertypes.CreatePoolMsg) (poolId uint64, err error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	poolId, err = server.keeper.swaprouterKeeper.CreatePool(ctx, msg)
+	if err != nil {
+		return 0, err
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.TypeEvtPoolCreated,
+			sdk.NewAttribute(types.AttributeKeyPoolId, strconv.FormatUint(poolId, 10)),
+		),
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeySender, msg.PoolCreator().String()),
+		),
+	})
+
+	return poolId, nil
+}
+
+func (server msgServer) CreateConcentratedPool(goCtx context.Context, msg *concentrated.MsgCreateConcentratedPool) (*concentrated.MsgCreateConcentratedPoolResponse, error) {
+	poolId, err := server.CreatePool(goCtx, msg)
+	if err != nil {
```

Comment:
I believe I wired it this way so that we can both call CreateConcentratedPool directly from the CL module, but its wired to CreatePool as well so that way the swaprouter module can also call it. I could be understanding this incorrectly though, so please lmk if you still feel otherwise.

Reply:
If I do that, then that means I can get rid of the MsgCreatorServer completely right?

Reply:
Yeah without the current CreateConcentratedPool then there is no point to the MsgCreatorServer since we are strictly creating the pool with a swaproutertypes.CreatePoolMsg.

Reply:
So does this mean decision made to remove this method?

Reply:
I think it wouldn't hurt to keep the method. Wdyt @p0mvn

Reply:
Why would we have to get rid of `MsgCreatorServer`?

`MsgCreatorServer` specifies a [`CreateConcentratedPool`](https://github.com/osmosis-labs/osmosis/blob/3735a4c99460a5b786df8764e8d3c317be90b5b1/x/concentrated-liquidity/model/tx.pb.go#L218) interface. 

As far as I can tell, there is no interface requiring `CreatePool`.

Therefore, we should be able to remove the old `CreateConcentratedPool` and rename the current `CreatePool` to `CreateConcentratedPool`.

Reply:
Yeah this was the same question I was referring to! ACK for me on this!